### ISSUE TITLE
Nfe 4.0

### ISSUE THREAD

**Issue by leandrodax** [thiagopena/djangoSIGE#34]
Parabéns pelo excelente projeto, porém gostaria de saber se já existe um previsão de atualização da biblioteca para versão 4.0 da Nfe??

**Comment by thiagopena** [comment]
Obrigado @leandrodax , o layout da NF-e 4.0 está implementado do PySIGNFe, mas falta adicionar ao processador e adicionar nas views do SIGE.

**Comment by thiagopena** [comment]
@leandrodax pode sim, essa parte está um pouco parada e não deve ser muito complexa de concluir. Seria uma grande ajuda essa implementação.

**Comment by robertowiest** [comment]
@leandrodax você conseguiu adicionar a NFe v4?

**Comment by leandrodax** [comment]
Não, tentei mas não consegui resolver erros na integração!!

**Comment by robertowiest** [comment]
Pois é, eu até me proponho a implementar, mas como não sou expert em python precisaria de algumas dicas de quais passos devo seguir.

**Comment by victtorvpb** [comment]
@robertowiest  se você decidir tocar essa empreitada posso ajudar...

**Comment by robertowiest** [comment]
Sim claro...podemos...mas por onde começo :D

**Comment by victtorvpb** [comment]
@robertowiest tem que atualizar a biblioteca de NF-e  ou substituir ela por uma que já foi portada para a versão 4.0, se o @thiagopena concordar com a substituição.

**Comment by thiagopena** [comment]
bom dia @victtorvpb @robertowiest , se a solução for migrar para outra lib estou de acordo. Mas acredito que o esforço seria menor de adaptar a PySIGNFe para NFe 4.0, pois os layouts já estão montados e já está integrada na aplicação, do que encontrar outra lib que implementa NFe 4.0 e reintegrar no código do SIGE.

De qualquer modo apoio qualquer decisão que melhore o projeto, mesmo que signifique não utilizar uma dependência minha hehe.

**Comment by robertowiest** [comment]
Certo, pelo que entendi então deixar funcional a PySIGNFe para v4. O que já foi feito e o que falta fazer?

**Comment by thiagopena** [comment]
@robertowiest o layout foi feito, falta integrar no processador de notas

**Comment by robertowiest** [comment]
Seria nesse arquivo somente https://github.com/thiagopena/PySIGNFe/blob/master/pysignfe/nfe/processador_nfe.py e criar um arquivo de webservices para a versão 4
?

**Comment by leandrodax** [comment]
Bom dia, quem tiver mais experiência com python oriente-nos, estou pronto para ajudar na implementação!!

**Comment by victtorvpb** [comment]
@leandrodax  e @robertowiest  tem uma issues no OySigNFe sobre a portabilidade para a versão 4.0, podemos conversar sobre essa portabilidade la. https://github.com/thiagopena/PySIGNFe/issues/3 depois de portarmos a biblioteca integramos a nova versão no DjangoSIGE.

**Comment by robertowiest** [comment]
Olá Pessoal, alguma novidade quanto a isso?

**Comment by mayconMello** [comment]
Bom dia @thiagopena 

As funcionalidades de envio de manifestação e distribuição estão contemplando a versão 4.0 ?

Eu queria aplicar em um projeto que preciso realizar download de notas fiscais.
